pkg/signing/signing.go

```diff
@@ -89,15 +87,64 @@ func sifAddSignature(fimg *sif.FileImage, groupid, link uint32, fingerprint [20]
 	return nil
 }
 
+// Copy-paste from sylabs/sif
+// datatypeStr returns a string representation of a datatype.
+func datatypeStr(dtype sif.Datatype) string {
+	switch dtype {
+	case sif.DataDeffile:
+		return "Def.FILE"
+	case sif.DataEnvVar:
+		return "Env.Vars"
+	case sif.DataLabels:
+		return "JSON.Labels"
+	case sif.DataPartition:
+		return "FS"
+	case sif.DataSignature:
+		return "Signature"
+	case sif.DataGenericJSON:
+		return "JSON.Generic"
+	case sif.DataGeneric:
+		return "Generic/Raw"
+	}
+	return "Unknown data-type"
+}
+
+func getDataPartitionToSign(fimg *sif.FileImage, dataType sif.Datatype) ([]*sif.Descriptor, error) {
+	sylog.Debugf("Looking for: %s partition to sign...", datatypeStr(dataType))
+	data, _, err := fimg.GetLinkedDescrsByType(uint32(0), dataType)
```

Comment:
could you please add a comment here explaning the logic behind the first argument?

It's looking for descriptors linking to ID 0 with a specific data type, so much is clear, but why 0? That's the bit that needs documenting so that other people don't get stuck on this.